vgg16_interpretation_vis.ipynb

```diff
@@ -0,0 +1,261 @@
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Zg3twKQyMVTe",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "from keras.applications.vgg16 import VGG16\n",
+        "from vis.visualization import visualize_activation\n",
+        "from vis.utils import utils\n",
+        "from keras.models import Sequential, Model\n",
+        "from keras.layers import Dense, Dropout, Activation, Flatten\n",
+        "from keras import activations\n",
+        "import matplotlib.pyplot as plt"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "XR7wj5UoHnzQ",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Step 2 First, let's understand the ImageNet dataset. Below, we are downloading descriptions for each of the 1000 classes of images in ImageNet. The descriptions can also be viewed [here.](https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "X8ITEW_ADrxm",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "import pickle\n",
+        "from urllib.request import urlopen\n",
+        "classidx_to_description_dict = pickle.load(urlopen('https://gist.githubusercontent.com/yrevar/6135f1bd8dcf2e0cc683/raw/d133d61a09d7e5a3b36b8c111a8dd5c4b5d560ee/imagenet1000_clsid_to_human.pkl'))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YWC5I8XiCCnm",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Step 3 Download and load the pre-trained model. For this exercise, we are using the [VGG16](https://arxiv.org/abs/1409.1556) architecture with weights pre-trained on [ImageNet](http://www.image-net.org) dataset. Note that in the last layer (predictions dense layer), the model output is classified into 1000 classes."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Mt_py1fjuLpK",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "model = VGG16()\n",
+        "model.summary()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "QcPATVq_Lh-E",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Step 4 Use activation maximization to visualize the images that maximize output of each filter in the prediction layer. Prediction layer is the layer in the model with 1000 filters, each representing a separate class of dataset.\n",
+        "### Question: What features has the model learnt for class *Indian Cobra* and *Baseball*, separately? How about other classes? Share with your table the interesting visualizations you get!"
```

Comment:
volcano might be a better example than indian cobra in terms of visualization. Can we change it?